Source/WebCore/page/LocalDOMWindow.cpp

```diff
@@ -2898,7 +2914,7 @@ ExceptionOr<RefPtr<WindowProxy>> LocalDOMWindow::open(LocalDOMWindow& activeWind
     if (!frame)
         return RefPtr<WindowProxy> { nullptr };
 
-    if (!firstWindow.allowPopUp()) {
+    if (!firstWindow.allowPopUp(*firstFrame, firstFrame->protectedDocument()->completeURL(urlString))) {
```

Comment:
I think you're doing more work than you should here for the normal case where someone doesn't need this quirk.

I suggest something like:

```
URL targetURL = firstWindow.protectedDocument()->quirks().needsPopupFromMicrosoftOfficeToOneDrive() ? firstFrame->protectedDocument()->completeURL(urlString) : {};
if (!firstWindow.allowPopUp(*firstFrame, WTFMove(targetURL)) {
    ...
```

Note: `Document::completeURL` returns a new copy, so we shouldn't call it if we do not need to.